### PR TITLE
Add settings to ignore second display name in search

### PR DESCRIPTION
**Pull request by artonge**
When LDAP backend is used and a second Display name property is set, the display name value will look like that: "First Display Name (Second Display Name)". This break user search when autocomplete is deactivated.

This PR adds a setting to match the first display name with a totally not bulletproof regexp.

Fix https://github.com/nextcloud-gmbh/customer-feature-requests/issues/6

### CODE REVIEW

File: lib/private/Collaboration/Collaborators/UserPlugin.php [hunk 1]
```diff
@@ -181,6 +184,7 @@ public function search($search, $limit, $offset, ISearchResult $searchResult) {
 				$this->shareeEnumerationFullMatch &&
 				$lowerSearch !== '' && (strtolower($uid) === $lowerSearch ||
 				strtolower($userDisplayName) === $lowerSearch ||
+				(strtolower(preg_replace('/ (.*)$/', '', $userDisplayName)) === $lowerSearch && $shareeEnumerationIgnoreSecondDisplayName) ||
```
Comment (PVince81) [inline lib/private/Collaboration/Collaborators/UserPlugin.php:187]: maybe swap the conditions so we don't run the regexp code if the flag is disabled (conditions are executed from left to the right)

**Review by PVince81 — COMMENTED** [review]
Looks nice. Can you also add a unit test ? I see we already have a Collaborators/UserPluginTest

**Comment by artonge** [comment]
> Looks nice. Can you also add a unit test ? I see we already have a Collaborators/UserPluginTest

Done

**Review by PVince81 — APPROVED** [review]
Nice, thanks a lot! :+1:

**Review by blizzz — CHANGES_REQUESTED** [review]

File: lib/private/Collaboration/Collaborators/UserPlugin.php [hunk 2]
```diff
@@ -181,6 +184,7 @@ public function search($search, $limit, $offset, ISearchResult $searchResult) {
 				$this->shareeEnumerationFullMatch &&
 				$lowerSearch !== '' && (strtolower($uid) === $lowerSearch ||
 				strtolower($userDisplayName) === $lowerSearch ||
+				($this->shareeEnumerationFullMatchIgnoreSecondDisplayName && strtolower(preg_replace('/ \(.*\)$/', '', $userDisplayName)) === $lowerSearch) ||
```
[the remark this answers is not in the document]
Comment (artonge) [reply lib/private/Collaboration/Collaborators/UserPlugin.php:187]: Yes, why not :)

[the remark this answers is not in the document]
Comment (artonge) [reply lib/private/Collaboration/Collaborators/UserPlugin.php]: Done

**Review by blizzz — APPROVED** [review]

**Pull request merged**